sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -202,3 +196,67 @@ case class CreateViewCommand(
     )
   }
 }
+
+/**
+ * Alter a view with given query plan. If the view name contains database prefix, this command will
+ * alter a permanent view matching the given name, or throw an exception if view not exist. Else,
+ * this command will try to alter a temporary view first, if view not exist, try permanent view
+ * next, if still not exist, throw an exception.
+ *
+ * @param name the name of this view.
+ * @param originalText the original SQL text of this view. Note that we can only alter a view by
+ *                     SQL API, which means we always have originalText.
+ * @param query the logical plan that represents the view; this is used to generate a canonicalized
+ *              version of the SQL that can be saved in the catalog.
+ */
+case class AlterViewCommand(
+    name: TableIdentifier,
+    originalText: String,
+    query: LogicalPlan) extends RunnableCommand {
+
+  override protected def innerChildren: Seq[QueryPlan[_]] = Seq(query)
+
+  override def run(session: SparkSession): Seq[Row] = {
+    // If the plan cannot be analyzed, throw an exception and don't proceed.
+    val qe = session.sessionState.executePlan(query)
+    qe.assertAnalyzed()
+    val analyzedPlan = qe.analyzed
+
+    val tableMeta = session.sessionState.catalog.getTableMetadata(name)
+    if (tableMeta.tableType != CatalogTableType.VIEW) {
+      throw new AnalysisException(s"${tableMeta.identifier} is not a view.")
+    }
+
+    if (name.database.isDefined) {
+      alterPermanentView(session, analyzedPlan)
+    } else if (session.sessionState.catalog.isTemporaryTable(name)) {
```

Comment:
It can be simplified to
```
if (session.sessionState.catalog.isTemporaryTable(name)) {
  session.sessionState.catalog.createTempView(name.table, analyzedPlan, overrideIfExists = true)
} else {
  alterPermanentView(session, analyzedPlan)
}
```

I am fine if you keep the existing way.